src/mtcp/mtcp_restart.c

```diff
@@ -243,8 +244,12 @@ MTCP_PRINTF("Attach for debugging.");
     }
     // This assumes that the MTCP header signature is unique.
     do {
-      mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
-    } while (mtcp_strcmp(mtcpHdr.signature, MTCP_SIGNATURE) != 0);
+      rc = mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
+    } while (rc > 0 && mtcp_strcmp(mtcpHdr.signature, MTCP_SIGNATURE) != 0);
+    if (rc == 0) { /* if end of file */
+      MTCP_PRINTF("***ERROR: ckpt image doesn't match MTCP_SIGNATURE\n");
```

Comment:
> I wonder if we should rename `MTCP_SIGNATURE` with `DMTCP_SIGNATURE` in the error message so as to not confuse the user with MTCP/DMTCP.

It's a good point, but here we're referring to the name in the
code:  MTCP_SIGNATURE
In the code, we distinguish between the DMTCP header, which comes first,
followed by the MTCP header.  So MTCP_SIGNATURE reflects the code better,
and hopefully, users will recognize the resemblance in the names.

By the way, what does LGTH mean?  (I like systems, because there are
always new things to learn.)

Best,
- Gene

On Wed, Mar 25, 2015 at 08:08:14AM -0700, Kapil Arya wrote:
> > @@ -243,8 +244,12 @@ MTCP_PRINTF("Attach for debugging.");
> >      }
> >      // This assumes that the MTCP header signature is unique.
> >      do {
> > -      mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
> > -    } while (mtcp_strcmp(mtcpHdr.signature, MTCP_SIGNATURE) != 0);
> > +      rc = mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
> > +    } while (rc > 0 && mtcp_strcmp(mtcpHdr.signature, MTCP_SIGNATURE) != 0);
> > +    if (rc == 0) { /* if end of file */
> > +      MTCP_PRINTF("***ERROR: ckpt image doesn't match MTCP_SIGNATURE\n");
> 
> I wonder if we should rename `MTCP_SIGNATURE` with `DMTCP_SIGNATURE` in the error message so as to not confuse the user with MTCP/DMTCP.
> 
> We should keep this in mind for other such error strings as well.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/dmtcp/dmtcp/pull/46/files#r27129426